crates/red_knot_python_semantic/src/node_key.rs

```diff
@@ -1,12 +1,18 @@
-use ruff_python_ast::{AnyNodeRef, NodeKind};
+use ruff_python_ast::{AnyNodeRef, Identifier, NodeKind};
 use ruff_text_size::{Ranged, TextRange};
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+pub(super) enum Kind {
+    Node(NodeKind),
+    Identifier,
+}
+
 /// Compact key for a node for use in a hash map.
 ///
 /// Compares two nodes by their kind and text range.
 #[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
 pub(super) struct NodeKey {
-    kind: NodeKind,
+    kind: Kind,
```

Comment:
I want to change the struct name because it doesn't correspond to only nodes now that an identifier can be represented as a key.

Reply:
For all I'm concerned, `Identifier` is a node. It has a value and range. It only lacks a `NodeKind` and a `AnyNode` implementation, but we could easily add that.

Reply:
Sounds good, I can add that as a follow-up.